core/src/main/java/io/aiven/kafka/tieredstorage/RemoteStorageManager.java

```diff
@@ -415,6 +469,8 @@ public InputStream fetchIndex(final RemoteLogSegmentMetadata remoteLogSegmentMet
             }
             final DetransformFinisher detransformFinisher = new DetransformFinisher(detransformEnum);
             return detransformFinisher.toInputStream();
+        } catch (final RemoteResourceNotFoundException e) {
+            throw e;
```

Comment:
This is redundant

Reply:
not really. As the throw is within a try-catch, is captured by the last `catch Exception` and turned into a RSMException -- breaking the exception handling on the broker side.

Reply:
Yes, you're right!